library/ajax/template_context_search.php

```diff
@@ -0,0 +1,44 @@
+<?php
+/**
+ * Ajax interface for custom template context.
+ *
+ * @package   OpenEMR
+ * @link      http://www.open-emr.org
+ * @author    Jerry Padgett <sjpadgett@gmail.com>
+ * @copyright Copyright (c) 2019 Jerry Padgett <sjpadgett@gmail.com>
+ * @license   https://github.com/openemr/openemr/blob/master/LICENSE GNU General Public License 3
+ */
+
+$cq = <<< createQuery
+CREATE OR REPLACE VIEW zView_user_templates AS
+SELECT cl2.cl_list_slno context_id, cl2.cl_list_item_long context_text,
+cl3.cl_list_slno cat_id, cl3.cl_list_item_long cat_text, cl3.cl_creator cat_ownerid,
+IFNULL(cl4.cl_list_slno,"") tmpl_id, IFNULL(cl4.cl_list_item_long,"") tmpl_text, IFNULL(cl4.cl_creator,"") tmpl_ownerid
+FROM customlists cl2
+INNER JOIN customlists cl3 on cl2.cl_list_slno=cl3.cl_list_id
+LEFT OUTER JOIN customlists cl4 on cl3.cl_list_slno=cl4.cl_list_id and cl4.cl_list_type=4 and cl4.cl_deleted=0
+WHERE cl2.cl_list_type=2 and cl2.cl_deleted=0
+and cl3.cl_list_type=3 and cl3.cl_deleted=0
+GROUP BY context_id;
+createQuery;
```

Comment:
To save a nanosecond/nanobit or 2, prob better to place this under the below globals.php include. So, if that fails (auth or something), then not needlessly creating the above parameter.

Reply:
Would even place it under the csrf stuff, for same reason as above.

Reply:
I believe I did this this way to throw your sense of uniformity out of whack!:)
For the escaping, I promised, one month.:)

Reply:
- If the project is ok with use of views, shouldn't it follow standard process of database.sql and upgrade?
- @sjpadgett, we use 'z' prefix as a reminder for localized objects and it got in the post.  If this is first view, may be it should establish a pattern for others to follow.

Reply:
I use a view in the portal for tracking activities and at the time I was led to believe the project wants to keep the database strictly tables.  I suppose one could consider a view as temporary and dynamic so keeping the way I have would be fine. For version control and uniformity, adding to database is appropriate but in the end, I vote to do them dynamic in code. However, I don't have a strong opinion.
For naming convention I used xxxx_view in portal but once again, i'll go with the flow here as well.

Reply:
We have been avoiding views/triggers etc. Does this view basically get rebuilt every time it is used? If this were the first view, I'd be nervous, but there is a view is in the portal, which also seems to get rebuilt every time it is used (fyi, I didn't even realize there was a view in the portal :) ). Are there any drawbacks to dynamic views like this?